### PR TITLE
Add initial Wikipedia integration

### PR DESCRIPTION
**Pull request by wclausen**

### CODE REVIEW

File: app/card_generation/people_getter.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        self.known_for_html = known_for_html
+
+
+def get_people(highlight: str) -> List[Person]:
+    result = []
+    doc = NLP(highlight)
+    for ent in doc.ents:
+        if ent.label_ == "PERSON":
+            result += [Person(ent.text)]
+    return result
```
Comment (z1lc) [inline app/card_generation/people_getter.py:41]: You may be able to replace this with `[Person(ent.text) for ent in NLP(highlight).ents if ent.label_ == "PERSON"]`. Happy to leave as-is though if we think we'll want to add more stuff in the `if`/`for`.

File: app/card_generation/people_getter.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+
+def get_wiki_page(name: str) -> Optional[WikipediaPage]:
+    try:
+        if len(wikipedia.search(name)) == 0:
```
Comment (z1lc) [inline app/card_generation/people_getter.py:46]: `if not wikipedia.search(name):`

File: app/card_generation/people_getter.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            # probably a typo, try with auto-suggest
+            return wikipedia.page(name, auto_suggest=True)
+        else:
+            # force no auto suggest
+            return wikipedia.page(name, auto_suggest=False)
+    except Exception as e:
+        log("Failed to find person in Wikipedia for: {name}".format(name=Person.name))
```
Comment (z1lc) [inline app/card_generation/people_getter.py:53]: Use can use string interpolation here: `f"Failed to find person in Wikipedia for: {Person.name}"`

File: app/card_generation/people_getter.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    return re.sub("(" + "|".join(sentence_starters) + ")", "", sentence, count=1)
+
+
+def remove_jr_sr(summary: str) -> str:
+    no_jr = summary.replace('Jr.', '')
+    no_jr_no_sr = no_jr.replace('Sr.', '')
+    return no_jr_no_sr
+
+
+def remove_name_refs(sentence: str, name: str) -> str:
```
Comment (z1lc) [inline app/card_generation/people_getter.py:81]: Can you add a comment to show what this is doing? Not sure I follow.

File: app/card_generation/readwise.py [hunk 5]
```diff
@@ -1,18 +1,30 @@
...
 
 READWISE_HIGHLIGHT_CLOZE_MODEL_ID = 1604800000000
 
 
+def generate_readwise_people(user: User, deck: Deck, tags: List[str]):
+    highlights = get_highlights(user)
+    all_people: Set[Person] = set()
+    for highlight in highlights:
+        people_in_highlight = get_people(highlight['text'])
+        all_people.union(set(people_in_highlight))
```
Comment (z1lc) [inline app/card_generation/readwise.py:22]: can you use `all_people.update(people_in_highlight)`?

File: tests/test_get_people.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        people = [Person(name) for name in famous_names]
+        wikipedia_people = [get_wikipedia_info(person) for person in people]
+        assert(len(list(filter(lambda person: person is None, wikipedia_people))) == 0)
+
+    def test_corrects_basic_typo(self):
+        typo_name = Person("Lance Armstrog")
+        real_name = Person("Lance Armstrong")
+        assert(get_wikipedia_info(typo_name) == get_wikipedia_info(real_name))
+
+    def test_lebron_html_sensible(self):
```
Comment (z1lc) [inline tests/test_get_people.py:27]: This test in particular I am concerned about -- I think we would expect the Wikipedia text here to change pretty quickly. Could you change it such that it's a less-sensitive match? Maybe eg add in `"basketball player" in html` and count that there's three `<li>` elements.

File: tests/test_get_people.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+        expected_known_for_html = "<ul>\n" + \
+            "<li>American professional basketball player for the Los Angeles Lakers of the National Basketball Association</li>\n" + \
+            "<li>frequently compared to Michael Jordan in debates over the greatest basketball player of all time</li>\n" + \
+            "<li>only player in NBA history to have brought NBA championships to three franchises as Finals MVP</li>" + \
+            "</ul>"
+        assert(get_wikipedia_info(Person("Lebron James")).known_for_html == expected_known_for_html)
+
+    def test_elon_html_sensible(self):
```
Comment (z1lc) [inline tests/test_get_people.py:35]: Similar comment to above.

**Review by z1lc — APPROVED** [review]
This looks like it's actually going to work really well! Main thing is making the integration tests a little less fragile.

File: app/card_generation/people_getter.py (hunk 1 above)
Comment (wclausen) [reply app/card_generation/people_getter.py:41]: 👍  nah, let's simplify it. no plans rn for future extension here

File: app/card_generation/people_getter.py (hunk 2 above)
Comment (wclausen) [reply app/card_generation/people_getter.py:46]: 👍

File: app/card_generation/people_getter.py (hunk 3 above)
Comment (wclausen) [reply app/card_generation/people_getter.py:53]: 👍

File: app/card_generation/people_getter.py (hunk 4 above)
Comment (wclausen) [reply app/card_generation/people_getter.py:81]: 👍

File: app/card_generation/readwise.py (hunk 5 above)
Comment (wclausen) [reply app/card_generation/readwise.py:22]: 👍

File: tests/test_get_people.py (hunk 6 above)
Comment (wclausen) [reply tests/test_get_people.py:27]: I'll refactor to have it use static text from LeBron's page today and verify it produces sensible html

File: tests/test_get_people.py (hunk 7 above)
Comment (wclausen) [reply tests/test_get_people.py:35]: refactored to pull in static summary text

**Pull request merged**